model-server-api/src/commonMain/kotlin/org/modelix/model/server/api/v2/VersionDeltaStreamV2.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.server.api.v2
+
+import io.ktor.http.ContentType
+import io.ktor.utils.io.ByteReadChannel
+import io.ktor.utils.io.ByteWriteChannel
+import io.ktor.utils.io.readUTF8Line
+import io.ktor.utils.io.writeStringUtf8
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class VersionDeltaStreamV2(
```

Comment:
Let's add some documentation to the format and a rationale why it is needed. Maybe, most of the implementation between v1 and v2 could also be shared apart from the final marker being sent.

Reply:
Sure. I will prepare an ADR for tomorrow's refinement.
It will be documentation and and also explain, why we use or do not reuse the old implementation.

Reply:
I'd also be fine with a comment on that class ;-)

Reply:
Ah, ok. I do not want to send the hashes like in the old version. So do not want to reuse the old implementation. I guess this is still something we should discuss in the refinement.

Reply:
I've also seen that regex once and was confused. So yes, if we can avoid the hashes, then that's fine for me. But I wouldn't have had to ask if this was documented somewhere ;-)

Reply:
Not sending the hashes is how it was implemented before. And then there was a bug in the JS library for computing SHA256 hashes that took me days to debug (see https://github.com/modelix/modelix.core/pull/213). Data integrity is more important than performance.

Reply:
Good know. I will add it again.

Reply:
Done. Added comment about rational for VersionDeltaStreamV2. Also mentioned why we still send hashes.